x-pack/plugins/cases/public/components/all_cases/use_all_cases_state.test.tsx

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { act, renderHook } from '@testing-library/react-hooks';
+
+import { TestProviders } from '../../common/mock';
+import {
+  useAllCasesState,
+  getQueryParamsLocalStorageKey,
+  getFilterOptionsLocalStorageKey,
+} from './use_all_cases_state';
+import { DEFAULT_FILTER_OPTIONS, DEFAULT_QUERY_PARAMS } from '../../containers/use_get_cases';
+import { stringify } from 'query-string';
+import { DEFAULT_TABLE_ACTIVE_PAGE, DEFAULT_TABLE_LIMIT } from '../../containers/constants';
+import { CaseStatuses } from '../../../common';
+import { SortFieldCase } from '../../containers/types';
+
+const LOCAL_STORAGE_QUERY_PARAMS_DEFAULTS = {
+  perPage: DEFAULT_QUERY_PARAMS.perPage,
+  sortOrder: DEFAULT_QUERY_PARAMS.sortOrder,
+};
+
+const LOCAL_STORAGE_FILTER_OPTIONS_DEFAULTS = {
+  severity: DEFAULT_FILTER_OPTIONS.severity,
+  status: DEFAULT_FILTER_OPTIONS.status,
+};
+
+const URL_DEFAULTS = {
+  ...DEFAULT_QUERY_PARAMS,
+  ...LOCAL_STORAGE_FILTER_OPTIONS_DEFAULTS,
+};
+
+const mockLocation = { search: '' };
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useLocation: jest.fn().mockImplementation(() => {
+    return mockLocation;
+  }),
+  useHistory: jest.fn().mockReturnValue({
+    replace: jest.fn(),
+    push: jest.fn(),
+    location: {
+      search: '',
+    },
+  }),
+}));
+
+const APP_ID = 'testAppId';
+const LOCALSTORAGE_QUERY_PARAMS_KEY = getQueryParamsLocalStorageKey(APP_ID);
+const LOCALSTORAGE_FILTER_OPTIONS_KEY = getFilterOptionsLocalStorageKey(APP_ID);
+
+describe('useAllCasesQueryParams', () => {
+  beforeEach(() => {
+    localStorage.clear();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('calls setState with default values on first run', () => {
+    const { result } = renderHook(() => useAllCasesState(), {
+      wrapper: ({ children }) => <TestProviders>{children}</TestProviders>,
+    });
+
+    expect(result.current.queryParams).toStrictEqual(DEFAULT_QUERY_PARAMS);
+    expect(result.current.filterOptions).toStrictEqual(DEFAULT_FILTER_OPTIONS);
+  });
+
+  it('updates localstorage with default values on first run', () => {
+    expect(localStorage.getItem(LOCALSTORAGE_QUERY_PARAMS_KEY)).toStrictEqual(null);
+    expect(localStorage.getItem(LOCALSTORAGE_FILTER_OPTIONS_KEY)).toStrictEqual(null);
+
+    renderHook(() => useAllCasesState(), {
+      wrapper: ({ children }) => <TestProviders>{children}</TestProviders>,
+    });
+
+    expect(JSON.parse(localStorage.getItem(LOCALSTORAGE_QUERY_PARAMS_KEY) ?? '{}')).toMatchObject({
```

Comment:
nit: if you want to avoid the coalescing I think we can use `!` since these are tests and if it is undefined the test will fail anyway right?